src/ast.rs

```diff
@@ -0,0 +1,1617 @@
+#[cfg(test)]
```

Comment:
Might be nice to clean all of this up (maybe generate some stuff via macros?) but seemed desirable to define the `*Unresolved` AST types next to the corresponding AST types